src/core/toga/widgets/base.py

```diff
@@ -168,7 +174,8 @@ def app(self, app):
         self._set_app(app)
 
     def _set_app(self, app):
-        pass
+        if app is not None:
+            app.widgets.add(self)
```

Comment:
As with `_set_window()`, this should be in the base setter, not the _set_app() call. This method only exists so that subclasses can extend the setter; there's no expectation that you'll need to call `super()._set_app()`.